core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingCurves.kt

```diff
@@ -106,6 +90,95 @@ fun addBrakingCurvesAtEOAs(
     return builder.build()
 }
 
+private fun computeMinSvlEoaIndCurve(
+    fullIndicationCurveEoa: EnvelopePart,
+    fullIndicationCurveSvl: EnvelopePart,
+    targetPosition: Double
+): Envelope {
+    // SvL indication curve should have positions before target position, and speeds above the
+    // release speed. Otherwise, follow EoA indication curve.
+    if (
+        fullIndicationCurveSvl.beginPos >= targetPosition ||
+            fullIndicationCurveSvl.maxSpeed <= NATIONAL_RELEASE_SPEED
+    ) {
+        return Envelope.make(fullIndicationCurveEoa)
+    }
+    val releaseSpeedPositionEoa = fullIndicationCurveEoa.interpolatePosition(NATIONAL_RELEASE_SPEED)
+    val releaseSpeedPositionSvl = fullIndicationCurveSvl.interpolatePosition(NATIONAL_RELEASE_SPEED)
+    val lastIndicationPart =
+        fullIndicationCurveEoa.sliceWithSpeeds(
+            releaseSpeedPositionEoa,
+            NATIONAL_RELEASE_SPEED,
+            fullIndicationCurveEoa.endPos,
+            fullIndicationCurveEoa.endSpeed
+        )
+    val slicedIndicationCurveEoa =
+        fullIndicationCurveEoa.sliceWithSpeeds(
+            fullIndicationCurveEoa.beginPos,
+            fullIndicationCurveEoa.beginSpeed,
+            releaseSpeedPositionEoa,
+            NATIONAL_RELEASE_SPEED
+        )
+    val slicedIndicationCurveSvl =
+        fullIndicationCurveSvl.sliceWithSpeeds(
+            fullIndicationCurveSvl.beginPos,
+            fullIndicationCurveSvl.beginSpeed,
+            releaseSpeedPositionSvl,
+            NATIONAL_RELEASE_SPEED
+        )
+    val startIntersectingRange =
+        max(slicedIndicationCurveEoa.beginPos, slicedIndicationCurveSvl.beginPos)
+    var refCurve = slicedIndicationCurveEoa
+    var otherCurve = slicedIndicationCurveSvl
+    if (
+        slicedIndicationCurveEoa.interpolateSpeed(startIntersectingRange) >
+            slicedIndicationCurveSvl.interpolateSpeed(startIntersectingRange)
+    ) {
+        refCurve = slicedIndicationCurveSvl
+        otherCurve = slicedIndicationCurveEoa
+    }
+    val pointCount = refCurve.pointCount()
```

Comment:
Idea (feel free to resolve by yourself once decided what to do):
```suggestion
    // TODO: uneeded for now: interate over points of both curves to improve precision
    val pointCount = refCurve.pointCount()
```

Reply:
See comment above: another PR incoming managing this issue directly :)